src/Illuminate/Cache/FileStore.php

```diff
@@ -169,9 +169,11 @@ protected function ensurePermissionsAreCorrect($path)
     /**
      * Increment the value of an item in the cache.
      *
+     * @template  TValue
+     *
      * @param  string  $key
      * @param  mixed  $value
-     * @return int
+     * @return int|\Illuminate\Support\HigherOrderTapProxy|TValue
```

Comment:
`tap()` only returns the proxy if the second parameter is not passed, which is not the case here

Reply:
How did you come to the conclusion that it does?